test/flycheck-test.el

```diff
@@ -3881,6 +3882,29 @@ Why not:
     (flycheck-ert-should-syntax-check
      "language/rust/lib-main/src/main.rs" 'rust-mode)))
 
+(flycheck-ert-def-checker-test rust-cargo rust notes
+  ;; We only get notes on the first build. Ensure we start from a
+  ;; clean directory each time.
+  (shell-command "cargo clean")
```

Comment:
Prefer `call-process` to `shell-command`, as there's no need to create a shell here.

Also, any idea of the current working directory when that command is called?  I think you can pass `--manifest-path` to make sure we are cleaning the right folder :)